src/utils/govPayMetadata.ts

```diff
@@ -11,8 +11,11 @@ type FormattedMetadata = NonNullable<GovUKCreatePaymentPayload["metadata"]>;
 
 const ITP_KEY = "paidViaInviteToPay" as const;
 
-const isPassportValue = (value: GovPayMetadataValue) =>
-  typeof value === "string" && value.startsWith("@");
+const isPassportValue = (
+  value: GovPayMetadataValue,
+  type: GovPayMetadata["type"],
+): value is string =>
+  typeof value === "string" && (value.startsWith("@") || type === "data");
```

Comment:
This is a temporary measure to allow both methods of checking for data values to work. Post-data migration, this can be simplified to just a check for the `type` property.